docs/architecture/adr-043-nft-module.md

```diff
@@ -0,0 +1,413 @@
+# ADR 43: NFT Module
+
+## Changelog
+
+- 05.05.2021: Initial Draft
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR defines the `x/nft` module which as a generic implementation of the NFT API, roughly "compatible" with ERC721.
+
+## Context
+
+NFTs are more digital assets than only crypto arts, which is very helpful for accruing value to the Cosmos ecosystem. As a result, Cosmos Hub should implement NFT functions and enable a unified mechanism for storing and sending the ownership representative of NFTs as discussed in https://github.com/cosmos/cosmos-sdk/discussions/9065.
+
+As was discussed in [#9065](https://github.com/cosmos/cosmos-sdk/discussions/9065), several potential solutions can be considered:
+
+- irismod/nft and modules/incubator/nft
+- CW721
+- DID NFTs
+- interNFT
+
+Since NFTs functions/use cases are tightly connected with their logic, it is almost impossible to support all the NFTs' use cases in one Cosmos SDK module by defining and implementing different transaction types.
+
+Considering generic usage and compatibility of interchain protocols including IBC and Gravity Bridge, it is preferred to have a generic NFT module design which handles the generic NFTs logic.
+
+This design idea can enable composability that application-specific functions should be managed by other modules on Cosmos Hub or on other Zones by importing the NFT module.
+
+The current design is based on the work done by [IRISnet team](https://github.com/irisnet/irismod/tree/master/modules/nft) and an older implementation in the [Cosmos repository](https://github.com/cosmos/modules/tree/master/incubator/nft).
+
+## Decision
+
+We will create a module `x/nft`, which contains the following functionality:
+
+- Store and transfer NFTs utilizing `x/bank`.
+- Mint and burn NFTs.
+- Use `x/authz` to implement ERC721-style authorization.
+- Query NFTs and their supply information.
+
+### Types
+
+#### Metadata
+
+We define a `Metadata` model for `NFT Type`, which is comparable to an ERC721 contract on Ethereum, under which a collection of `NFT`s can be created and managed.
+
+```protobuf
+message Metadata {
+  string type          = 1; // Required, unique key, alphanumeric
+  string name          = 2;
+  string symbol        = 3;
+  string description   = 4;
+  bool mint_restricted = 5;
+  bool edit_restricted = 6;
+}
+```
+
+- The `type` is the identifier of the NFT type/class.
+- The `name` is a descriptive name of this NFT type.
+- The `symbol` is the symbol usually shown on exchanges for this NFT type.
```

Comment:
why NFT type / class needs a symbol? Could you update the PR and provide a bit more background?